edgelet/edgelet-core/src/module.rs

```diff
@@ -379,6 +379,66 @@ impl SystemInfo {
     }
 }
 
+#[derive(Debug, serde_derive::Serialize)]
+pub struct SystemResources {
+    host_uptime: u64,
+    process_uptime: u64,
+    used_cpu: f64,
+    used_ram: u64,
+    total_ram: u64,
+    disks: Vec<DiskInfo>,
+    docker_stats: String,
+}
+
+impl SystemResources {
+    pub fn new(
+        host_uptime: u64,
+        process_uptime: u64,
+        used_cpu: f64,
+        used_ram: u64,
+        total_ram: u64,
+        disks: Vec<DiskInfo>,
+        docker_stats: String,
+    ) -> Self {
+        SystemResources {
+            host_uptime,
+            process_uptime,
+            used_cpu,
+            used_ram,
+            total_ram,
+            disks,
+            docker_stats,
+        }
+    }
+}
+
+#[derive(Debug, serde_derive::Serialize)]
+pub struct DiskInfo {
```

Comment:
Since you've added this to `-core`, can you remove it from `iotedge/src/check/additional_info.rs` ?

Reply:
I was planning on making additional info use this path instead of doing it itself, but it doesn't make sense in this pr. I'll make a separate issue to do it, but probably mark it as low priority

Reply:
What do you mean by "use this path" ? If you mean you were going to have `iotedge check` invoke the `/resources` management endpoint to get the system resources info, then you should definitely not do that. Otherwise `iotedge check` won't be able to collect system resources information unless `iotedged` is up and running.

Reply:
I mean having check call runtime.system_resources() in check. It won't go through the http endpoint.

Reply:
Although, now that I look at it the endpoint gets a lot more info, like docker and cpu use. check shouldn't get this, so it should probably be kept separate. 

To address your original comment, the two disk info classes are different, since the check diskinfo stores human readable info (10.6MB) instead of just numbers.